src/legacy/ui/ui_exports/ui_export_types/saved_object.js

```diff
@@ -30,10 +30,28 @@ export const mappings = wrap(
   flatConcatAtType
 );
 
+const pluginId = (pluginSpec) => pluginSpec.id ? pluginSpec.id() : pluginSpec.getId();
+
 // Combines the `migrations` property of each plugin,
-// ensuring that properties are unique across plugins.
+// ensuring that properties are unique across plugins
+// and has migrations defined where the mappings are defined.
 // See saved_objects/migrations for more details.
-export const migrations = wrap(alias('savedObjectMigrations'), uniqueKeys(), mergeAtType);
+export const migrations = wrap(
+  alias('savedObjectMigrations'),
+  (next) => (acc, spec, type, pluginSpec) => {
+    const mappings = pluginSpec._uiExports.mappings || {};
```

Comment:
Please use the `getExportSpecs()` method instead of accessing the pseudo-private `_uiExports` property